src/json.ts

```diff
@@ -8,3 +9,20 @@ export function parseJsonWithLog<T>(result: string): T {
     throw e;
   }
 }
+
+// Use this method when parsing input coming from the [ShellRunner] so the stderr can be potentially shown.
+export function parseJsonStdoutWithLog<T>(result: ShellOutput): T {
+  try {
+    return JSON.parse(result.stdout) as T;
+  } catch (e) {
+    if (result.stdout === "") {
```

Comment:
f: I think this is not so easy. Only because stdout is empty must not mean an error has occurred here.
You must check for `result.code` != 0 and the existence of `result.stderr` to determine if there was an error.

Reply:
I'll improve that.